src/main/java/com/yahoo/bullet/bql/extractor/FilterExtractor.java

```diff
@@ -85,17 +93,43 @@ protected Clause visitLogicalBinaryExpression(LogicalBinaryExpression node, Void
 
         @Override
         protected Clause visitComparisonExpression(ComparisonExpression node, Void context) {
-            return createFilterClause(node.getOperation(), node.getLeft().toFormatlessString(), node.getRight().toFormatlessString());
+            Expression left = node.getLeft();
+            Operation op = node.getOperation();
+            boolean isNotExpression = false;
+            if (left instanceof ReferenceWithFunction) {
+                if (op != EQUALS && op != NOT_EQUALS) {
```

Comment:
I know we don't support SIZEOF(*) with >, < but is that possible to do? We need changes to bullet-core as well.

Reply:
Ah ok, maybe we should rename SIZEOF to SIZEOFEQUALS or something to be explicit

Reply:
I guess we can leave SIZEOF here. And we may extend it to support >, < and other operations and don't need to change the gramma once we do the change in bullet-core.

Reply:
I meant in bullet-core actually and the only way I see to support SIZE OF > etc is to allow another op to passed with SIZEOF, which I am not sure we'll be supporting anytime soon. Ok, let's leave it for now